src/models/timeaxisheaderview.cpp

```diff
@@ -0,0 +1,257 @@
+#include "timeaxisheaderview.h"
+
+#include <QPainter>
+#include <QtMath>
+#include <QDebug>
+
+#include "filterandzoomstack.h"
+
+TimeAxisHeaderView::TimeAxisHeaderView(FilterAndZoomStack *filterAndZoomStack, Qt::Orientation orientation, QWidget *parent)
+    : QHeaderView(orientation, parent)
+    , m_filterAndZoomStack(filterAndZoomStack)
+{
+    setMinimumHeight(40);
+
+    connect(filterAndZoomStack, &FilterAndZoomStack::filterChanged, this, &TimeAxisHeaderView::onFilterChanged);
+    connect(filterAndZoomStack, &FilterAndZoomStack::zoomChanged, this, &TimeAxisHeaderView::onZoomChanged);
+}
+
+void TimeAxisHeaderView::setTimeRange(const Data::TimeRange &timeRange)
+{
+    m_timeRange = timeRange;
+    emit headerDataChanged(orientation(), 1, 1);
+}
+
+void TimeAxisHeaderView::onFilterChanged(const Data::FilterAction &filterAction)
```

Comment:
this could be merged with the onZoomChanged handler, and both could be moved into a lambda. I.e. in the ctor write:

```
auto emitHeaderDataChanged = [this]() {
    emit headerDataChanged(orientation(), 1, 1);
};
connect(filterAndZoomStack, &FilterAndZoomStack::filterChanged, this, emitHeaderDataChanged);
connect(filterAndZoomStack, &FilterAndZoomStack::zoomChanged, this, emitHeaderDataChanged);
```